terminus/tests/circle_test.py

```diff
@@ -32,6 +34,33 @@ def test_intersection_at_one_point_in_nested_circles(self):
         self.assertEqual(circle1.intersection(circle2), [Point(4, 0)])
 
     def test_intersection_at_two_points(self):
+        # with each center outside the other circle
         circle1 = Circle(Point(-3, 0), 5)
         circle2 = Circle(Point(3, 0), 5)
         self.assertEqual(circle1.intersection(circle2), [Point(0, 4), Point(0, -4)])
+        # with one center inside the other circle
+        circle1 = Circle(Point(0, 0), math.sqrt(65))
+        circle2 = Circle(Point(4, 0), 5)
+        self.assertAlmostEqual(circle1.intersection(circle2), [Point(7.0, 4.0, 0.0), Point(7.0, -4.0, 0.0)])
+        circle1 = Circle(Point(0, 0), 5)
```

Comment:
It's a little difficult to explain without a drawing, but I'll try to write a nice comment for it. I has to do with the cases of the tests that don't pass. This case is in the border of the cases I had considered and the cases I hadn't.